### PR TITLE
create admin site and write testcases for models and views

### PR DESCRIPTION
**Pull request by hitansivekariya33**

### CODE REVIEW

File: Django_Blog/blog/templates/blog/blog_list.html [hunk 1]
```diff
@@ -37,7 +37,9 @@
             <h6><span>Publish at:</span> {{blog.created_at}}</h6>
         </div><br>
     {% endfor %} 
-
+    <div class="card">
+        <button class="btn btn-warning" type="submit"><a href="{% url 'blogform' %}">Add Blog</a></button>
+    </div>
     {% comment %} -------craete pagination---------- {% endcomment %}
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/templates/blog/blog_list.html]: Remove this comment.

File: Django_Blog/blog/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    def test_name_max_length(self):
+        author = Author.objects.get(id=1)
+        max_length = author._meta.get_field('name').max_length
+        self.assertEqual(max_length,100)
+
+    #testcase for check author object name
+    def test_object_name(self):
+        author = Author.objects.get(id=1)
+        get_object_name = author.name
+        self.assertEqual(get_object_name,str(author))
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_models.py]: Is this test case passed?

File: Django_Blog/blog/tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+class BlogModelTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        author = Author.objects.create(name='Jennifer Dasal',about_author = 'Jennifer Dasal is the curator of modern and contemporary art at the North Carolina Museum of Art in Raleigh, North Carolina, and the host of the independent podcast ArtCurious, which she started in 2016 and which was named one of the best podcasts by O, The Oprah Magazine and PC Magazine. She holds an MA in art history from the University of Notre Dame and a BA in art')
+        Blog.objects.create(title='Test Blog', author=author ,blog_content='This is a test blog.')
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_models.py]: Please remove the space after `author=author`. You can use the black formatter to format python file.

File: Django_Blog/blog/tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    def test_title_max_length(self):
+        blog = Blog.objects.get(id=1)
+        max_length = blog._meta.get_field('title').max_length
+        self.assertEqual(max_length,100)
+    
+    #testcase for check blog object name
+    def test_object_name(self):
+        blog = Blog.objects.get(id=1)
+        get_object_name = blog.title
+        self.assertEqual(get_object_name,str(blog))
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_models.py]: Also create test case to check for the blog content.

File: Django_Blog/blog/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    #testcase for check title maximum length 
+    def test_title_max_length(self):
+        blog = Blog.objects.get(id=1)
+        max_length = blog._meta.get_field('title').max_length
+        self.assertEqual(max_length,100)
+    
+    #testcase for check blog object name
+    def test_object_name(self):
+        blog = Blog.objects.get(id=1)
+        get_object_name = blog.title
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_models.py]: Rename this to `title` as it is not blog name.

File: Django_Blog/blog/tests/test_models.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    #testcase for check comment content label
+    def test_comment_content_label(self):
+        comment = Comment.objects.get(id=1)
+        field_label = comment._meta.get_field('comment_content').verbose_name
+        self.assertEqual(field_label, 'comment content')
+
+    #testcase for check comment content object name
+    def test_object_name(self):
+        comment = Comment.objects.get(id=1)
+        get_object_name = comment.comment_content
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_models.py]: Rename the variable name to `comment_content`. It is not a name.

File: Django_Blog/blog/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            blog_author = Author.objects.create(
+                name = 'Jennifer Dasal',
+                about_author = 'Jennifer Dasal is the curator of modern and contemporary art at the North Carolina Museum of Art in Raleigh, North Carolina, and the host of the independent podcast ArtCurious, which she started in 2016 and which was named one of the best podcasts by O, The Oprah Magazine and PC Magazine. She holds an MA in art history from the University of Notre Dame and a BA in art')
+            
+            Blog.objects.create(
+                title = f'Test Blog {blog_num}',
+                author = blog_author,
+                blog_content = f'Test Blog {blog_num}')
+
+    #The BlogListView (all-blog page) is accessible at the expected location (e.g./blog/blogs)
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_views.py]: Give space after `#`

File: Django_Blog/blog/tests/test_views.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'blog/blog_list.html')
+
+    #The BlogListView pagination records by 5 (at least on the first page)
+    def test_pagination_is_five(self):
+        response = self.client.get(reverse('blogs'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['blog_list']), 5)
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/tests/test_views.py]: You have added 6 blogs. So please add one more test function that test the `number of blogs = 6`.

File: Django_Blog/blog/tests/test_models.py (hunk 2 above)
Comment (hitansivekariya33) [reply Django_Blog/blog/tests/test_models.py]: Yes

File: Django_Blog/blog/urls.py [hunk 9]
```diff
@@ -1,10 +1,12 @@
...
+from .views import IndexView,AuthorListView,AuthorDetailView,BlogListView,BlogDetailView,commentview,AuthorForm,BlogForm
 urlpatterns = [
     path('',IndexView.as_view(),name='index'),
     path('blogger/',AuthorListView.as_view(),name='authorlist'),
     path('blogger/<int:pk>/',AuthorDetailView.as_view(),name='authordetail'),
-    path('blogs/',BlogListView.as_view(),name='bloglist'),
+    path('blogger/form/',AuthorForm.as_view(),name='authorform'),
+    path('blogs/',BlogListView.as_view(),name='blogs'),
     path('blogs/<int:pk>/',BlogDetailView.as_view(),name='blogdetail'),
     path('blogs/<int:pk>/create/',commentview,name='comment'),
```
Comment (himanshu-suthar-tagline) [inline Django_Blog/blog/urls.py]: Why this commentview is in small case. Change it to upper case.
Also please install the `black` formatter and format all the python files using it.

**Pull request merged**